timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceIntegrationTest.java

```diff
@@ -190,6 +206,106 @@ public void canGetCurrentTimeMillis() {
+        LockRefreshToken token = lockWithFullResponse(requestForWriteLock(LOCK_A), TEST_CLIENT).getLockRefreshToken();
+        Set<LockRefreshToken> lockRefreshTokens = cluster.lockService().refreshLockRefreshTokens(
+                ImmutableList.of(token));
+        assertThat(lockRefreshTokens).contains(token);
+    }
+
+    @Test
+    public void unlockAndFreezeDoesNotAllowRefreshes() throws InterruptedException {
+        lockWithFullResponse(requestForWriteLock(LOCK_A), TEST_CLIENT);
+        HeldLocksToken token2 = lockWithFullResponse(requestForWriteLock(LOCK_A), TEST_CLIENT);
+        cluster.lockService().unlockAndFreeze(token2);
+        Set<LockRefreshToken> lockRefreshTokens = cluster.lockService().refreshLockRefreshTokens(
+                ImmutableList.of(token2.getLockRefreshToken()));
+        assertThat(lockRefreshTokens).isEmpty();
+    }
+
+    @Test
+    public void getAndRefreshTokens() throws InterruptedException {
+        lockWithFullResponse(requestForWriteLock(LOCK_A), TEST_CLIENT);
+        lockWithFullResponse(requestForWriteLock(LOCK_B), TEST_CLIENT);
+
+        Set<HeldLocksToken> tokens = cluster.lockService().getTokens(TEST_CLIENT);
+        assertThat(tokens.stream()
+                .map(token -> Iterables.getOnlyElement(token.getLocks()).getLockDescriptor())
+                .collect(Collectors.toList())).containsExactlyInAnyOrder(LOCK_A, LOCK_B);
+
+        Set<HeldLocksToken> heldLocksTokens = cluster.lockService().refreshTokens(tokens);
+        assertThat(heldLocksTokens.stream()
+                .map(token -> Iterables.getOnlyElement(token.getLocks()).getLockDescriptor())
+                .collect(Collectors.toList())).containsExactlyInAnyOrder(LOCK_A, LOCK_B);
+    }
+
+    @Test
+    public void lockGrantsCanBeServedAndRefreshed() throws InterruptedException {
+        HeldLocksToken heldLocksToken = lockWithFullResponse(requestForReadLock(LOCK_A), TEST_CLIENT);
+        HeldLocksGrant heldLocksGrant = cluster.lockService().convertToGrant(heldLocksToken);
+
+        assertThat(cluster.lockService().getTokens(TEST_CLIENT)).isEmpty();
+
+        HeldLocksGrant refreshedLockGrant = cluster.lockService().refreshGrant(heldLocksGrant);
+        assertThat(refreshedLockGrant).isEqualTo(heldLocksGrant);
+
+        HeldLocksGrant refreshedLockGrant2 = cluster.lockService().refreshGrant(heldLocksGrant.getGrantId());
+        assertThat(refreshedLockGrant2).isEqualTo(heldLocksGrant);
+
+        cluster.lockService().useGrant(TEST_CLIENT_2, heldLocksGrant);
+        assertThat(Iterables.getOnlyElement(cluster.lockService().getTokens(TEST_CLIENT_2)).getLockDescriptors())
+                .contains(LOCK_A);
+
+        assertThatThrownBy(() -> cluster.lockService().useGrant(TEST_CLIENT_3, heldLocksGrant.getGrantId()))
+                .isInstanceOf(AtlasDbRemoteException.class);
+    }
+
+    @Test
+    public void getMinLockedInVersionIdWorks() throws InterruptedException {
```

Comment:
Could this have a more descriptive name? Looks like its checking the behavior when there is no min version id